synapse/lib/stormtypes.py

```diff
@@ -6546,6 +6547,132 @@ def __init__(self, runt, valu, path=None):
+                   )}},
+
+        {'name': 'iter', 'desc': 'Yield (<path>, <valu>) tuples for the users JSON objects.',
+         'type': {'type': 'function', '_funcname': 'iter',
+                  'args': (
+                       {'name': 'path', 'type': 'str|list', 'desc': 'A path string or list of path parts.'},
+                       {'name': 'prop', 'type': 'str|list', 'desc': 'A property name or list of name parts.', 'default': None},
+                   ),
+                   'returns': {'name': 'Yields', 'type': 'list', 'desc': '(<path>, <item>) tuples.'}}},
+    )
+
+    def __init__(self, runt, valu):
+        Prim.__init__(self, valu)
+        self.runt = runt
+        self.locls.update({
+            'get': self.get,
+            'set': self.set,
+            'has': self.has,
+            'del': self._del,
+            'iter': self.iter,
+        })
+
+    async def has(self, path):
+
+        path = await toprim(path)
+        if isinstance(path, str):
+            path = tuple(path.split('/'))
+
+        fullpath = ('users', self.valu, 'json') + path
+        if self.runt.user.iden != self.valu:
+            self.runt.confirm(('user', 'json', 'get'))
+
+        return await self.runt.snap.core.hasJsonObj(fullpath)
+
+    async def get(self, path, prop=None):
+        path = await toprim(path)
+        prop = await toprim(prop)
+
+        if isinstance(path, str):
+            path = tuple(path.split('/'))
+
+        fullpath = ('users', self.valu, 'json') + path
+
+        if self.runt.user.iden != self.valu:
+            self.runt.confirm(('user', 'json', 'get'))
+
+        if prop is None:
+            return await self.runt.snap.core.getJsonObj(fullpath)
+
+        return await self.runt.snap.core.getJsonObjProp(fullpath, prop=prop)
+
+    async def set(self, path, valu, prop=None):
+        path = await toprim(path)
+        valu = await toprim(valu)
+        prop = await toprim(prop)
+
+        if isinstance(path, str):
+            path = tuple(path.split('/'))
+
+        fullpath = ('users', self.valu, 'json') + path
+
+        if self.runt.user.iden != self.valu:
+            self.runt.confirm(('user', 'json', 'set'))
+
+        if prop is None:
+            await self.runt.snap.core.setJsonObj(fullpath, valu)
+            return True
+
+        return await self.runt.snap.core.setJsonObjProp(fullpath, prop, valu)
+
+    async def _del(self, path, prop=None):
+        path = await toprim(path)
+        prop = await toprim(prop)
+
+        if isinstance(path, str):
+            path = tuple(path.split('/'))
+
+        fullpath = ('users', self.valu, 'json') + path
+
+        if self.runt.user.iden != self.valu:
+            self.runt.confirm(('user', 'json', 'set'))
+
+        if prop is None:
+            await self.runt.snap.core.delJsonObj(fullpath, valu)
+
+        return await self.runt.snap.core.delJsonObjProp(fullpath, valu, prop=prop)
+
+    async def iter(self):
```

Comment:
What do you think about having an optional path arg here?  I was considering namespacing internal usage like "synapse/<pkg/svc name>/foo".